webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx

```diff
@@ -1,106 +1,139 @@
 export const ADMIN_LABEL = 'Admin';
-export const API_SUCCESS_MESSAGE = `New user with empty password has been created, 
+export const API_SUCCESS_MESSAGE_CREATE = `New user with empty password has been created, 
 advice user to reset password as soon as possible.`;
-
+export const API_SUCCESS_MESSAGE_EDIT = 'User has been updated';
 interface Props {
   branding: BrandingType;
   form: FormInstance;
+  user?: DetailedUser
 }
 
-const ModalForm: React.FC<Props> = ({ form, branding }) => (
-  <Form
-    form={form}
-    labelCol={{ span: 8 }}
-    wrapperCol={{ span: 14 }}>
-    <Form.Item
-      label={USER_NAME_LABEL}
-      name={USER_NAME_NAME}
-      required
-      rules={[
-        {
-          message: 'Please type in your username.',
-          required: true,
-        },
-      ]}
-      validateTrigger={[ 'onSubmit' ]}>
-      <Input autoFocus maxLength={128} placeholder="user name" />
-    </Form.Item>
-    <Form.Item
-      label={DISPLAY_NAME_LABEL}
-      name={DISPLAY_NAME_NAME}>
-      <Input maxLength={128} placeholder="display name" />
-    </Form.Item>
-    {branding === BrandingType.Determined ? (
+const ModalForm: React.FC<Props> = ({ form, branding, user }) => {
+  useEffect(() => {
+    form.setFieldsValue({
+      [ADMIN_NAME]: user?.isAdmin,
+      [DISPLAY_NAME_NAME]: user?.displayName,
+    });
+  }, [ user, form ]);
+  return (
+    <Form
+      form={form}
+      labelCol={{ span: 8 }}
+      wrapperCol={{ span: 14 }}>
+      <Form.Item
+        initialValue={user?.username}
+        label={USER_NAME_LABEL}
+        name={USER_NAME_NAME}
+        required
+        rules={[
+          {
+            message: 'Please type in your username.',
+            required: true,
+          },
+        ]}
+        validateTrigger={[ 'onSubmit' ]}>
+        <Input autoFocus disabled={!!user} maxLength={128} placeholder="user name" />
+      </Form.Item>
       <Form.Item
-        label={ADMIN_LABEL}
-        name={ADMIN_NAME}>
-        <Switch defaultChecked={false} />
+        label={DISPLAY_NAME_LABEL}
+        name={DISPLAY_NAME_NAME}>
+        <Input maxLength={128} placeholder="display name" />
       </Form.Item>
-    ) : null }
-  </Form>
-);
+      {branding === BrandingType.Determined ? (
+        <Form.Item
+          label={ADMIN_LABEL}
+          name={ADMIN_NAME}
+          valuePropName="checked">
+          <Switch />
+        </Form.Item>
+      ) : null }
+    </Form>
+  );
+};
 
 interface ModalProps {
   onClose?: () => void;
+  user?: DetailedUser
 }
 
-const useModalCreateUser = ({ onClose }: ModalProps): ModalHooks => {
+const useModalCreateUser = ({ onClose, user }: ModalProps): ModalHooks => {
+  // const [ user, setUser ] = useState<DetailedUser>()
   const [ form ] = Form.useForm();
   const { info } = useStore();
-
+  const [ updatedUser, setUpdatedUser ] = useState<DetailedUser>();
   const { modalOpen: openOrUpdate, ...modalHook } = useModal();
 
+  const fetchUser = useCallback(async () => {
```

Comment:
Because useModal renders modal with `user` props, after editing, I was not able to update `user` associated with modal from user management page... (suppose I edit display name, the table can reflect that change because data can be refetched, but if I open edit modal again, without refreshing page, inside the modal it is still the old display name) I know this is inefficient, open to better ideas..

Reply:
i actually don't think i'm seeing the behavior you're describing when i test without `updatedUser` and `fetchUser`. it shows the updated values when i reopen the modal. 

which i think makes sense, since the `user` is sent from the page to the modal via the `ModalProps`, so if the page has the updated values, i'd assume the modal should as well. 

can you take a look at that part again? i may be missing something but i think we should be able to make this work in a way that the modal doesn't need to do it's own extra fetch after updating, if we're already refetching the list of users.

i'm also remembering we have a specific pattern for sending data from page to modal that was recently set up, so curious if @hamidzr has any thoughts on the `user` prop being used in `UserManagement.tsx` and `useModalCreateUser.tsx`

Reply:
You are absolutely right, thank you for catching this! 
After looking into it, I realize that the problem I was running into is due to I started using `initialValue` instead of `setFieldsValue`. Unnecessary fetch is removed, thank you~